### DISCUSSION TITLE
A sister application for PocketSpecs -- This one is CullFront -- for the frontend development

### DISCUSSION THREAD

**Discussion by square-foot** [pocketbase/pocketbase#6187] [Show and tell]
Sometime back I had pointed out PocketSpecs. (Docs at https://docs.pocketspecs.xyz) which helps me generate Pocketbase code quickly. 

I now coded a new kind of front-end generator. It is a "sister" application for PocketSpecs. You can "kaizen" into your HTML frontend/website with at least 40% less typing. It allows separation of concerns and you can code using HTMX and AlpineJS (or equivalent tech) Been planning this for a long time -- finally coded it on Christmas day. https://cullfront.com/ The website is sparse. But do read the documentation -- it explains everything.

I would be renaming PocketSpecs to CullCode -- as it would support other backends too (with Pocketbase being the main one) so CullCode and CullFront should work as a pair.

This is fresh off the oven. So kindly let me know your thoughts. Thank you for reading till here.

**Comment by nicosql** [comment]
I don't see the value of doing this unless you were porting an app that was already written and even then, geting it into your format is painstaking without any tooling.  My thoughts are this is not useful for greenfield apps but kudos for completing the project.

**Reply by square-foot** [reply]
Thank you for your feedback. I am not sure if you are speaking after using the tool/s. There is a one time difficulty (or uneasiness) in using these tools. But the difficulty does not grow.  The documentation is quite small for both.  I am already developing two complex SaaS applications with this (In fact that was the reason for making me invent this) They are private projects for two businesses, so cannot share . One is a SaaS for a travel agency. And the other is a hospital management SaaS. I didn't understand why anyone would want to re-make something else with this. Maybe I missed something in your message.
